docs/nmt-lib.md

```diff
@@ -0,0 +1,187 @@
+# Namespace Merkle Tree Library
+The Namespace Merkle Tree (NMT) library implements the NMT data structure outlined in the [NMT spec](./spec/nmt.md). 
+In the following sections, we will provide instructions on how to utilize the library to construct an NMT and offer insights into its fundamental methods.
+
+## NMT Initialization and Configuration
+
+An NMT can be constructed using the `New` function.
+```go
+func New(h hash.Hash, setters ...Option) *NamespacedMerkleTree
+```
+It receives a base hash function alongside with some optional configurations, namely:
+1. Namespace ID byte-size: If not specified then a default maximum is applied by the library.
```

Comment:
[question] is it a maximum or just a default byte size? IIUC a namespace id can't be less than 8 so it doesn't seem like a maximum

```suggestion
1. Namespace ID byte-size: If not specified then a default is applied by the library.
```

Reply:
Right, using the word `maximum` in here causes confusion, like you said, no ID can have any length less than the designated namespace ID size. It is now revised.

Reply:
Please see https://github.com/celestiaorg/nmt/pull/120/commits/9d50cafbc6dea994a29bc7a768e23d7c35106761